 ### Exploración y preparación de los datos

In [ ]:
%pip install https://github.com/pandas-profiling/pandas-profiling/archive/master.zip

In [1]:
def missing_values_table(df):
    mis_val = df.isnull().sum()
    mis_val_percent = 100 * df.isnull().sum() / len(df)
    mis_val_table = pd.concat([mis_val, mis_val_percent], axis=1)
    mis_val_table_ren_columns = mis_val_table.rename(
    columns = {0 : 'Missing Values', 1 : '% of Total Values'})
    mis_val_table_ren_columns = mis_val_table_ren_columns[
        mis_val_table_ren_columns.iloc[:,1] != 0].sort_values(
    '% of Total Values', ascending=False).round(1)
    print ("Your selected dataframe has " + str(df.shape[1]) + " columns.\n"      
        "There are " + str(mis_val_table_ren_columns.shape[0]) +
            " columns that have missing values.")
    return mis_val_table_ren_columns

In [2]:
# PARA FORMATEAR LOS FLOATS Y NO SE IMPRIMAN CON NOTACION CIENTÍFICA.
import pandas as pd
pd.set_option('display.float_format', lambda x: '%.5f' % x)

In [47]:
# librerías base
import pandas as pd
from IPython.display import HTML
from IPython.display import display
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

from IPython.display import HTML
from IPython.display import display

# cargamos el dataset
data = pd.read_excel('BaseDatosProyecto.xlsx')

#data = data.drop("Empresa", axis= 1)

numericalColumns = data._get_numeric_data().columns.tolist() 
categoricalColumns = list(set(data.columns) - set(data.describe().columns))

data.set_index('Empresa', inplace=True)

data.head()

print('Valores faltantes:')
display(missing_values_table(data)) 

print(data.shape)


Valores faltantes:
Your selected dataframe has 7 columns.
There are 0 columns that have missing values.


Empty DataFrame
Columns: [Missing Values, % of Total Values]
Index: []

(1381, 7)


### Perfilamiento de los datos

In [16]:
data.head()

Annual Turnover  Employee Count  Activos Fijos Aon Office  \
Empresa                                                              
132         579245735319            3038   716954431572     Bogota   
1061         67176109000             975    38508144000     Bogota   
622         154255168000            1587    87866173000     Bogota   
173        1271848954000             157   462891146000     Bogota   
144         433259253705             558   129907322353       Cali   

                                       Industry TIER GENERAL     Produccion  
Empresa                                                                      
132      Food System, Agribusiness and Beverage       TIER 1 -7258043.85604  
1061             Business and Personal Services       TIER 3 -4130887.93592  
622                Transportation and Logistics       TIER 3 -1822678.46122  
173                                      Energy       TIER 1 -1288362.12245  
144                Pharmaceutical and Chemicals       TIER 2 -1242213.71020

In [49]:
dataPreparada = pd.get_dummies(data, columns=["Aon Office", "Industry", "TIER GENERAL"])
dataPreparada.head()

Annual Turnover  Employee Count  Activos Fijos     Produccion  \
Empresa                                                                  
132         579245735319            3038   716954431572 -7258043.85604   
1061         67176109000             975    38508144000 -4130887.93592   
622         154255168000            1587    87866173000 -1822678.46122   
173        1271848954000             157   462891146000 -1288362.12245   
144         433259253705             558   129907322353 -1242213.71020   

         Aon Office_Barranquilla  Aon Office_Bogota  Aon Office_Cali  \
Empresa                                                                
132                            0                  1                0   
1061                           0                  1                0   
622                            0                  1                0   
173                            0                  1                0   
144                            0                  0                1   

         Aon Office_Medellin  Aon Office_TBD  Aon Office_TBD Colombia  ...  \
Empresa                                                                ...   
132                        0               0                        0  ...   
1061                       0               0                        0  ...   
622                        0               0                        0  ...   
173                        0               0                        0  ...   
144                        0               0                        0  ...   

         Industry_Professional Services  Industry_Public Sector  \
Empresa                                                           
132                                   0                       0   
1061                                  0                       0   
622                                   0                       0   
173                                   0                       0   
144                                   0                       0   

         Industry_Real Estate  Industry_Retail and Wholesale Trade  \
Empresa                                                              
132                         0                                    0   
1061                        0                                    0   
622                         0                                    0   
173                         0                                    0   
144                         0                                    0   

         Industry_Technology and Communications  \
Empresa                                           
132                                           0   
1061                                          0   
622                                           0   
173                                           0   
144                                           0   

         Industry_Transportation and Logistics  TIER GENERAL_TIER 1  \
Empresa                                                               
132                                          0                    1   
1061                                         0                    0   
622                                          1                    0   
173                                          0                    1   
144                                          0                    0   

         TIER GENERAL_TIER 2  TIER GENERAL_TIER 3  TIER GENERAL_TIER 4  
Empresa                                                                 
132                        0                    0                    0  
1061                       0                    1                    0  
622                        0                    1                    0  
173                        0                    0                    0  
144                        1                    0                    0  

[5 rows x 33 columns]

### Se cuenta con 8 variables. 3 de ellas son categóricas.  
### La base de datos tiene 1381 observaciones y carece de valores faltantes.
### Como parte de la preparación de los datos se ha decidido utilizar la variable Empresa como índice y transformar a una codificación numérica las variables categóricas __Aon Office, Industry, y Tier General__.  Esto aumenta la dimensionalidad hasta 33 variables.

In [51]:
import pandas_profiling
from pandas_profiling import ProfileReport

profile = ProfileReport(dataPreparada)

#profile.to_notebook_iframe()

profile.to_file("DatosPreparados.html")



Export report to file: 100%|██████████| 1/1 [00:00<00:00, 20.83it/s]
